### PR TITLE
Modified navbar.svelte such that menu will close when link clicked

### LINKED ISSUE #124: Menu does not close after page navigation 🧱
The mobile menu component currently does not close when a menu item is clicked. This requires the user to click the ❌ on the top right of the screen in order to close the menu.

The expected behavior, however, is to close the menu automatically when a menu item is clicked.

### PR DESCRIPTION
**Pull request by anhduy1202**
Hi, hope this is the correct way to make a PR, let me know if I need to improve anything in my code, thank you.

### CODE REVIEW

File: src/lib/components/sections/navbar.svelte [hunk 1]
```diff
@@ -1,42 +1,57 @@
 <script lang="ts">
 	export let segment: string;
+	let menu;
+	let checkbox;
```
Comment (EthanThatOneKid) [inline src/lib/components/sections/navbar.svelte]: This is just a nitpick, but to be as explicit with our intentions with a variable as possible in TypeScript, we can define variables with their expected HTML element. Like so:

```ts
let menu: HTMLUListElement;
let checkbox: HTMLInputElement;
```

File: src/lib/components/sections/navbar.svelte [hunk 2]
```diff
@@ -1,42 +1,57 @@
...
-			slug: 'nodebuds'
+			slug: 'nodebuds',
 		},
-		{ title: 'connect', slug: 'connect' }
+		{ title: 'connect', slug: 'connect' },
 	];
+
+	const handleClose = () => {
+		if (checkbox.checked === true) {
+			menu.style.visibility = 'hidden';
```
Comment (EthanThatOneKid) [inline src/lib/components/sections/navbar.svelte]: The neat thing about the technique used with the menu is that it does not actually need any JavaScript to tell it to hide/unhide; the menu toggling is handled by the checkbox input in the CSS. Line 20 can be omitted.

File: src/lib/components/sections/navbar.svelte [hunk 3]
```diff
@@ -1,42 +1,57 @@
...
+						aria-current={segment === slug}>
 						{@html title}
 					</a>
 				</li>
 			{/each}
 		</ul>
 	</div>
 
-	<label for="navToggle" class="menu">
+	<label for="navToggle" class="menu" on:click={() => handleOpen()}>
```
Comment (EthanThatOneKid) [inline src/lib/components/sections/navbar.svelte]: An `on:click` listener is not needed to open the menu since it is handled by the checkbox input.

**Review by EthanThatOneKid — CHANGES_REQUESTED** [review]
It is exciting to see the menu closing automatically now! I have a few nitpicks, but once those are resolved we can merge this PR.

File: src/lib/components/sections/navbar.svelte (hunk 3 above)
Comment (anhduy1202) [reply src/lib/components/sections/navbar.svelte]: Hi, I just changed it

File: src/lib/components/sections/navbar.svelte (hunk 2 above)
Comment (anhduy1202) [reply src/lib/components/sections/navbar.svelte]: Done!

File: src/lib/components/sections/navbar.svelte (hunk 1 above)
Comment (anhduy1202) [reply src/lib/components/sections/navbar.svelte]: Done :D

**Review by EthanThatOneKid — APPROVED** [review]

**Pull request merged**